### PR TITLE
Move wait timeout from queueset to APF filter

### PR DESCRIPTION
**Pull request by MikeSpreitzer**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
This PR is an alternate take on #120222 , replacing the explicit pruning of queues with request context expiration.  This take does the deadline shortening high up on the stack, at the point where the context is separated from the request.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by wojtek-t** [comment]
/assign @tkashem 

@tkashem  - Can you please take a look?
I skimmed through it and I like this approach, but I would like a review from you first.

**Comment by MikeSpreitzer** [comment]
@MikeSpreitzer

File: staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/interface.go [hunk 1]
```diff
@@ -129,10 +128,6 @@ type QueuingConfig struct {
 	// HandSize is a parameter of shuffle sharding.  Upon arrival of a request, a queue is chosen by randomly
 	// dealing a "hand" of this many queues and then picking one of minimum length.
 	HandSize int
-
```
Comment (wojtek-t) [inline staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/interface.go:132]: nit: update the comment for DesiredNumQueues above

**Review by wojtek-t — COMMENTED** [review]
I looked deeper into both PRs and they are fairly similar - given that @tkashem started working on that before, I suggest closing this one and proceed with Abu's.

@MikeSpreitzer - WDYT ^^

**Pull request closed without merging**